src/modules/status-list/status-list.service.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+} from '@nestjs/common';
+import { URL } from 'url';
+import { ConfigService } from '@nestjs/config';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { verifyCredential } from 'didkit-wasm-node';
+import { gzip } from 'pako';
+import {
+  CredentialWithStatusDto,
+  StatusList2021EntryDto,
+} from './dtos/credential-status.dto';
+import { STATUS_LIST_MODULE_PATH } from './status-list.const';
+import { CredentialDto } from './dtos/credential.dto';
+import { VerifiableCredentialDto } from './dtos/verifiable-credential.dto';
+import { StatusListCredentialDto } from './dtos/status-list-credential.dto';
+import { StatusListVerifiableCredentialDto } from './dtos/status-list-verifiable-credential.dto';
+import { StatusListCredential } from './entities/status-list-credential.entity';
+import { DID } from '../did/did.types';
+
+@Injectable()
+export class StatusListService {
+  constructor(
+    private readonly configService: ConfigService,
+    @InjectRepository(StatusListCredential)
+    private readonly statusListRepository: Repository<StatusListCredential>
+  ) {}
+
+  /**
+   * Add StatusList2021Entry as `credentialStatus` parameter to given credential object.
+   * https://w3c-ccg.github.io/vc-status-list-2021/#statuslist2021entry
+   *
+   * @param {CredentialDto} credential
+   * @return credential with StatusList2021Entry
+   */
+  async addStatusListEntry(
+    credential: CredentialDto
+  ): Promise<CredentialWithStatusDto> {
+    if (await this.isCredentialRevoked(credential.id)) {
+      throw new ConflictException('Credential already revoked');
+    }
+
+    const statusListCredential = new URL(
+      `${STATUS_LIST_MODULE_PATH}/${credential.id}`,
+      this.configService.get('STATUS_LIST_DOMAIN')
+    );
+
+    const newStatusList = StatusListCredential.create({
+      credentialId: credential.id,
+      namespace: credential.credentialSubject.role.namespace,
+    });
```

Comment:
To me, it would make more sense to use the `statusListCredential.id`, rather than the VC's id here. This would reduce the changes and confusion that would be necessary if the implementation changed to a "many VC to one status-list" relationship.
I guess a downside might be that we would need to use UUIDs for `StatusListCredential`, but I think this is worth it.
```suggestion
    const statusListCredentialId = uuid();
    const statusListCredentialUrl = new URL(
      `${STATUS_LIST_MODULE_PATH}/${statusListCredentialId}`,
      this.configService.get('STATUS_LIST_DOMAIN')
    );

    const newStatusList = StatusListCredential.create({
      id: statusListCredentialId,
      credentialId: credential.id,
      namespace: credential.credentialSubject.role.namespace,
    });
```